fastai/tabular/core.py

```diff
@@ -237,8 +237,18 @@ def _decode_cats(voc, c): return c.map(dict(enumerate(voc[c.name].items)))
 class Categorify(TabularProc):
     "Transform the categorical variables to something similar to `pd.Categorical`"
     order = 1
+    def __init__(self, classes=None):
+        store_attr()
+        super().__init__()
     def setups(self, to):
-        store_attr(classes={n:CategoryMap(to.iloc[:,n].items, add_na=(n in to.cat_names)) for n in to.cat_names})
+        if self.classes is None:
+            classes={n:CategoryMap(to.iloc[:,n].items, add_na=(n in to.cat_names)) for n in to.cat_names}
+        else:
+            classes = {}
```

Comment:
Would `defaultdict` allow us to simplify this?

Reply:
It will and it won't. There's some weird behavior with if it's a categorical datatype in the `DataFrame` (see next push), but considering we did it always anyways I went ahead and did a check for it to recalculate

Reply:
Wound up not working as it should the further we got in the pipeline (see recent comment)